File: Lecture003 If-Else, While loop & Lots of Patterns/Pattern31.java
package com.company;

import java.util.Scanner;

public class Pattern31 {
    // Pattern 31
    public static void main(String [] args){
        /*
        Pattern 31
                             4 4 4 4 4 4 4
                             4 3 3 3 3 3 4
                             4 3 2 2 2 3 4
                             4 3 2 1 2 3 4
                             4 3 2 2 2 3 4
                             4 3 3 3 3 3 4
                             4 4 4 4 4 4 4

         */
        // Creating Scanner Object For taking inputs from the USER
        Scanner sc = new Scanner(System.in);
        // Take Number of ROWS and COLUMNS from the USER
        System.out.print(" Enter Number of ROWS for the Pattern: ");
        int n = sc.nextInt();
        pattern(n);
    }
    static void pattern(int n){
        // Procedure to print the Required Pattern
        int rcSize = 2 * n - 1; // Row Column SIZE
        for(int row =1; row <=rcSize; row++){
            for(int column = 1; column <= rcSize; column++){
                int atEveryIndex = n -  Math.min(Math.min(row - 1, column - 1), Math.min(rcSize  - row , rcSize  - column));
                System.out.print(atEveryIndex + " ");
            }
            System.out.println();
        }
    }
}

File: 003_If-Else_While_loop_and_Lots_of_Patterns/Swift/patterns/Pattern31.swift

import Foundation

// Function to print the specified pattern
func printPattern(_ n: Int) {
    let rcSize = 2 * n - 1
    for row in 1...rcSize {
        for column in 1...rcSize {
            let atEveryIndex = n - min(min(row - 1, column - 1), min(rcSize - row, rcSize - column))
            print("\(atEveryIndex) ", terminator: "")
        }
        print("")
    }
}

// Taking user input for the number of rows
print("Enter Number of ROWS for the Pattern: ", terminator: "")
if let input = readLine(), let n = Int(input) {
    printPattern(n)
} else {
    print("Invalid input")
}
